pai/predictor.py

```diff
@@ -296,22 +294,65 @@ def delete_service(self):
         """Delete the service."""
         self.session.service_api.delete(name=self.service_name)
 
-    def wait_for_ready(self):
-        """Wait until the service enter running status."""
+    def wait_for_ready(self, force: bool = False):
+        """Wait until the service enter running status.
+
+        Args:
+            force (bool): Whether to force wait for ready.
+
+        Raises:
+            RuntimeError: Raise if the service terminated unexpectedly.
+
+        """
+        if self.service_status == ServiceStatus.Running and not force:
```

Comment:
I still think `force` should be used to check whether hacky wait for eas or not.
suggestion:
```
if self.service_status == ServiceStatus.Running:
    return
...
if force:
    self._wait_for_gateway_ready()
...
```